aging_due_report/wizard/wizard.py

```diff
@@ -347,26 +346,27 @@ def _get_lines_by_partner_without_invoice(self, rp_ids):
                 # TODO: This process can be improve by using the approach reach
                 # in commission_payment
                 date_due = [amx_brw.date_maturity or amx_brw.date
-                            for amx_brw in aml_lines
-                            if amx_brw.journal_id.type in ('sale', 'purchase')]
+                            for amx_brw in aml_lines.filtered(
+                                lambda r: r.journal_id.type in (
+                                    'sale', 'purchase'))]
                 date_due = (date_due and min(date_due) or
                             min([amy_brw.date_maturity or amy_brw.date
                                  for amy_brw in aml_lines]))
                 date_emission = [amz_brw.date
-                                 for amz_brw in aml_lines
-                                 if amz_brw.journal_id.type in (
-                                     'sale', 'purchase')]
+                                 for amz_brw in aml_lines.filtered(
+                                     lambda r: r.journal_id.type in (
+                                         'sale', 'purchase'))]
```

Comment:
You are using filtered (good!) but mapped is not used yet.

I mean, you could use:
`aml_lines.filtered(lambda r: r.journal_id.type in ('sale', 'purchase'))].mapped('date')`